plugins/node/opentelemetry-instrumentation-oracledb/test/oracle.test.ts

```diff
@@ -0,0 +1,1675 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (c) 2025, Oracle and/or its affiliates.
+ * */
+import {
+  Attributes,
+  SpanStatusCode,
+  context,
+  Span,
+  SpanKind,
+  SpanStatus,
+  trace,
+} from '@opentelemetry/api';
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+import * as testUtils from '@opentelemetry/contrib-test-utils';
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+  ReadableSpan,
+  TimedEvent,
+} from '@opentelemetry/sdk-trace-base';
+import * as assert from 'assert';
+import { OracleInstrumentation } from '../src';
+import { SpanNames, DB_SYSTEM_VALUE_ORACLE } from '../src/constants';
+
+import {
+  ATTR_NETWORK_TRANSPORT,
+  ATTR_SERVER_ADDRESS,
+  ATTR_SERVER_PORT,
+  ATTR_EXCEPTION_MESSAGE,
+  ATTR_EXCEPTION_STACKTRACE,
+  ATTR_EXCEPTION_TYPE,
+} from '@opentelemetry/semantic-conventions';
+
+import {
+  ATTR_DB_NAMESPACE,
+  ATTR_DB_SYSTEM,
+  ATTR_DB_STATEMENT,
+  ATTR_DB_OPERATION_PARAMETER,
+  ATTR_DB_USER,
+  ATTR_DB_OPERATION_NAME,
+} from '../src/semconv';
+
+const memoryExporter = new InMemorySpanExporter();
+let contextManager: AsyncHooksContextManager;
+const provider = new BasicTracerProvider({
+  spanProcessors: [new SimpleSpanProcessor(memoryExporter)],
+});
+const tracer = provider.getTracer('external');
+const instrumentation = new OracleInstrumentation();
+instrumentation.enable();
+instrumentation.disable();
+
+import * as oracledb from 'oracledb';
+
+const VER_23_4 = 2304000000;
+const hostname = 'localhost';
+const pno = 1521;
+const serviceName = 'FREEPDB1';
+let serverVersion = 2304000000; // DB version.
+let numExecSpans = 2; // Default number of Spans created for Execute API in thin mode.
+let numConnSpans = 2; // Default number of spans created during connection establishment.
+let poolMinSpanCount = 1; // number of spans created for createPool considering poolMin.
+const CONFIG = {
+  user: process.env.ORACLE_USER || 'demo',
+  password: process.env.ORACLE_PASSWORD || 'demo',
+  connectString: process.env.ORACLE_CONNECTSTRING || 'localhost:1521/freepdb1',
```

Comment:
```suggestion
```

Reply:
> 

This is a configuration attribute that contains information such as `host`, `port`, and `servicename`. It's required to establish a database connection and run the tests. However, it is never validated within the span, as it does not appear there.